IP.cc:202:3: error: ‘dump_this_packet’ was not declared in this scope

With -DENABLE_MOBILE_IPV6 compilation fails at 4.0.0 with:

`FAILED: src/packet_analysis/protocol/ip/CMakeFiles/plugin-PacketAnalyzer-IP.dir/IP.cc.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DDOCTEST_CONFIG_DISABLE -Isrc/packet_analysis/protocol/ip -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip -Isrc/packet_analysis -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis -Isrc -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src -Ilibkqueue-build/include -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/auxil/libkqueue/include -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/auxil/highwayhash -Iauxil/binpac/lib -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/auxil/binpac/lib -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/auxil/rapidjson/include -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/auxil/paraglob/include -Iauxil/broker/include -I/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/auxil/broker/include -I.  -march=native -O2 -Wall -Wno-unused -Wno-register -Werror=vla -std=c++17 -MD -MT src/packet_analysis/protocol/ip/CMakeFiles/plugin-PacketAnalyzer-IP.dir/IP.cc.o -MF src/packet_analysis/protocol/ip/CMakeFiles/plugin-PacketAnalyzer-IP.dir/IP.cc.o.d -o src/packet_analysis/protocol/ip/CMakeFiles/plugin-PacketAnalyzer-IP.dir/IP.cc.o -c /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip/IP.cc
/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip/IP.cc: In member function ‘virtual bool zeek::packet_analysis::IP::IPAnalyzer::AnalyzePacket(size_t, const uint8_t*, zeek::Packet*)’:
/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip/IP.cc:202:3: error: ‘dump_this_packet’ was not declared in this scope
  202 |   dump_this_packet = true;
      |   ^~~~~~~~~~~~~~~~
/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip/IP.cc:204:10: warning: ‘ignore_checksums’ is deprecated: Remove in v4.1. Use zeek::detail::ignore_checksums. [-Wdeprecated-declarations]
  204 |   if ( ! ignore_checksums && mobility_header_checksum(packet->ip_hdr) != 0xffff )
      |          ^~~~~~~~~~~~~~~~
In file included from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/iosource/Packet.h:18,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/packet_analysis/Manager.h:8,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/packet_analysis/Analyzer.h:4,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/packet_analysis/protocol/ip/IP.h:5,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip/IP.cc:3:
/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/NetVar.h:114:13: note: declared here
  114 | extern int& ignore_checksums [[deprecated("Remove in v4.1. Use zeek::detail::ignore_checksums.")]];
      |             ^~~~~~~~~~~~~~~~
/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip/IP.cc:204:63: error: cannot convert ‘std::unique_ptr<zeek::IP_Hdr>’ to ‘const zeek::IP_Hdr*’
  204 |   if ( ! ignore_checksums && mobility_header_checksum(packet->ip_hdr) != 0xffff )
      |                                                       ~~~~~~~~^~~~~~
      |                                                               |
      |                                                               std::unique_ptr<zeek::IP_Hdr>
In file included from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/Val.h:15,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/plugin/ComponentManager.h:9,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/packet_analysis/Manager.h:7,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/packet_analysis/Analyzer.h:4,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/packet_analysis/protocol/ip/IP.h:5,
                 from /var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/packet_analysis/protocol/ip/IP.cc:3:
/var/tmp/portage/net-analyzer/zeek-4.0.0/work/zeek-4.0.0/src/zeek/zeek/net_util.h:180:51: note:   initializing argument 1 of ‘int zeek::mobility_header_checksum(const zeek::IP_Hdr*)’
  180 | extern int mobility_header_checksum(const IP_Hdr* ip);
      |                                     ~~~~~~~~~~~~~~^~`

--- a/src/IP.cc
+++ b/src/IP.cc
@@ -245,7 +245,7 @@ RecordValPtr IPv6_Hdr::ToVal(VectorValPtr chain) const
 		case 5:
 			{
 			auto m = make_intrusive<RecordVal>(ip6_mob_bu_type);
-			m->Assign(0, ntohs(*((uint16_t*)msg_data))));
+			m->Assign(0, ntohs(*((uint16_t*)msg_data)));
 			m->Assign(1, static_cast<bool>(ntohs(*((uint16_t*)(msg_data + sizeof(uint16_t)))) & 0x8000));
 			m->Assign(2, static_cast<bool>(ntohs(*((uint16_t*)(msg_data + sizeof(uint16_t)))) & 0x4000));
 			m->Assign(3, static_cast<bool>(ntohs(*((uint16_t*)(msg_data + sizeof(uint16_t)))) & 0x2000));

--- a/src/packet_analysis/protocol/ip/IP.cc
+++ b/src/packet_analysis/protocol/ip/IP.cc
@@ -200,9 +200,9 @@ bool IPAnalyzer::AnalyzePacket(size_t len, const uint8_t* data, Packet* packet)
 	// last if present.
 	if ( packet->ip_hdr->LastHeader() == IPPROTO_MOBILITY )
 		{
-		dump_this_packet = true;
+		packet->dump_packet = true;
 
-		if ( ! ignore_checksums && mobility_header_checksum(packet->ip_hdr) != 0xffff )
+		if ( ! detail::ignore_checksums && mobility_header_checksum(packet->ip_hdr.get()) != 0xffff )
 			{
 			Weird("bad_MH_checksum", packet);
 			return false;

